### PR TITLE
New linear algebra algorithm

### PR DESCRIPTION
**Pull request by nic-dern**
Hey,
I added a new linear algebra algorithm which takes a number of x and y coordinates as input and then outputs a polynomial function that connects them.
Thank you for taking a look at it.

### CODE REVIEW

File: linear_algebra/src/python-polynom-for-points.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+from decimal import *
+getcontext()
+getcontext().prec=28
+
+count=0
+check=1
+zero_y=[]
+
+#number of points you want to use
```
Comment (cclauss) [inline linear_algebra/src/python-polynom-for-points.py]: Our automated tests fail:
```
==================================== ERRORS ====================================

_______ ERROR collecting linear_algebra/src/python-polynom-for-points.py _______

linear_algebra/src/python-polynom-for-points.py:11: in <module>

    x=int(input("number of points: ").strip())

../../../virtualenv/python3.7.1/lib/python3.7/site-packages/_pytest/capture.py:661: in read

    raise IOError("reading from stdin while output is captured")

E   OSError: reading from stdin while output is captured

------------------------------- Captured stdout --------------------------------

number of points: 
```
so please indent the following lines underneath `if __name__ == "__main__":`

Comment (nic-dern) [reply linear_algebra/src/python-polynom-for-points.py]: I tried to improve the code! Thank you for your quick response.

File: linear_algebra/src/python-polynom-for-points.py [hunk 2]
```diff
@@ -1,139 +1,98 @@
...
-            solved+="+"
-        count+=1
-
-    print(solved)
-
-elif check==2:
-    print(solved)
-else:
-    print("The program can not work out that function yet.")
+def points_to_polynomial(coordinates[][]):
```
Comment (cclauss) [inline linear_algebra/src/python-polynom-for-points.py]: At the top of the script, please put:
```python
from decimal import *


def points_to_polynomial(coordinates):
    # coordinates is a two dimensional matrix: [[x, y],[x, y],...]
```

File: linear_algebra/src/python-polynom-for-points.py [hunk 3]
```diff
@@ -1,139 +1,98 @@
...
+                if count+1 != x:
+                    solved+="+"
+                count+=1
+
+            return solved
+
+        elif check==2:
+            return solved
+        else:
+            return "The program cannot work out a fitting polynomial."
```
Comment (cclauss) [inline linear_algebra/src/python-polynom-for-points.py]: At the bottom of the script, please put:
```python

if __name__ == "__main__":
    points_to_polynomial([[1, 1], [2, 2], [3, 3]])  # or some better matrix
```

File: linear_algebra/src/python-polynom-for-points.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
+    
+from decimal import *
+getcontext().prec=28
+
+
+def points_to_polynomial(coordinates):
+    #the list (coordinates) is two dimensional: [[x, y],[x, y],...]
+
+    if __name__ == "__main__" :
```
Comment (cclauss) [inline linear_algebra/src/python-polynom-for-points.py]: Remove this line here and put it down at the bottom of the script OUTSIDE of __points_to_polynomial()__.

**Comment by cclauss** [comment]
Current behavior...

__python3 linear_algebra/src/python-polynom-for-points.py__
```python
The program cannot work out a fitting polynomial.
The program cannot work out a fitting polynomial.
Traceback (most recent call last):
  File "linear_algebra/src/python-polynom-for-points.py", line 107, in <module>
    print(points_to_polynomial([[1, 5], [2, 2], [3, 9]]))
  File "linear_algebra/src/python-polynom-for-points.py", line 35, in points_to_polynomial
    if check==1:
UnboundLocalError: local variable 'check' referenced before assignment
```

**Pull request merged**